pkg/api/types.go

```diff
@@ -705,6 +713,8 @@ type ServiceSpec struct {
 	// not be changed by updates.
 	PortalIP string `json:"portalIP,omitempty"`
 
+	Rewrite bool `json:"rewrite,omitempty" description:"Does the load balancer rewrite packets or pass them through"`
```

Comment:
Looking through the code, it looks like rewrite is only valid when createExternalLoadBalancer is true and PublicIPs is empty. Therefore, it appears it belongs in ServiceStatus. Errors for invalid uses would be useful, also.